crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -478,6 +484,48 @@ impl SpaceInternalHandler for Space {
         .into())
     }
 
+    fn handle_notify(&mut self, to_agent: KAgent, data: wire::Wire) -> InternalHandlerResult<()> {
+        let ro_inner = self.ro_inner.clone();
+        let timeout = ro_inner.config.tuning_params.implicit_timeout();
+
+        Ok(async move {
+            match discover::search_and_discover_peer_connect(
+                ro_inner.clone(),
+                to_agent.clone(),
+                timeout,
+            )
+            .await
+            {
+                discover::PeerDiscoverResult::OkShortcut => {
+                    use kitsune_p2p_types::codec::Codec;
+                    if let Err(err) = ro_inner
+                        .evt_sender
+                        .notify(
+                            ro_inner.space.clone(),
+                            to_agent,
+                            data.encode_vec().expect("encode data"),
```

Comment:
Notify itself is fire-and-forget, there's no guarantee it will be received on the remote, and even if it is, the remote could be swamped with response sends and decide to drop it.

This is why I want to manage the retry timing from the fetch side. Does that make sense?